webapp/src/main/java/com/box/l10n/mojito/okapi/filters/JSONFilter.java

```diff
@@ -94,6 +101,15 @@ public void handleValue(String value, JsonValueTypes valueType) {
         extractNoteIfMatch(value);
         extractUsageIfMatch(value);
         super.handleValue(value, valueType);
+        if (comment != null) {
+            addXliffNote(comment);
```

Comment:
can this be in the "handleComment" function?

Reply:
No, because when comment is being processed, textUnit is not yet created. TextUnit gets created in `handleValue`. The order of call is `handleComment`, `handleKey` and `handleValue` for the following code: 
```
{
  // Greeting from Main UI
  "hello": "Hello!"
}
```